src/shape/java/us/ihmc/euclid/shape/convexPolytope/interfaces/Face3DReadOnly.java

```diff
@@ -688,6 +690,35 @@ default boolean orthogonalProjection(Point3DReadOnly pointToProject, Point3DBasi
          return closestVisibleEdge.orthogonalProjection(pointToProject, projectionToPack);
    }
 
+   
+   /**
+    * Calculates the intersection between line and face3D
+    *
+    * @param pointOnLine   a random point on the line.
+    * @param lineDirection direction of the line.
+    * 
+    * @return whether the intersection exists or not.
+    */
+   
+
+   default boolean intersectionBetweenLine3DAndFace3D(Point3DReadOnly pointOnLine, Vector3DReadOnly lineDirection, Point3DBasics intersectionToPack)
+   {
+      boolean intersectionWithPlane = EuclidGeometryTools.intersectionBetweenLine3DAndPlane3D(getVertices().get(0), getNormal(), pointOnLine, lineDirection, intersectionToPack);
+      
+      if (intersectionWithPlane) {
+         
+         boolean insideTheFace = isPointInside(intersectionToPack, EuclidGeometryTools.ONE_MILLIONTH);
+         
+         if (insideTheFace)
+            return true;
+   
+      }
+      
+      return false;
```

Comment:
To be consistent with the other intersection algos, let's call `intersectionToPack.setToNaN()` when it's not intersecting.